charmcraft/parts/plugins/_python.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+"""Charmcraft-specific poetry plugin."""
+
+import shlex
+from pathlib import Path
+
+from craft_parts.plugins import python_plugin
+from overrides import override
+
+from charmcraft import utils
+
+
+class PythonPluginProperties(python_plugin.PythonPluginProperties, frozen=True):
+
+    python_requirements: list[str] = ["requirements.txt"]  # By default use requirements.txt
```

Comment:
seems like an awkward default, what if there isn't a 'requirements.txt' file?

Reply:
If there's not a requirements.txt file and the user uses the default it'll fail with that error.

The reason I chose this is that most existing charms (as well as the templates) use a `requirements.txt` file. The ones that don't are using poetry, so we'd want to move them to the poetry module instead. However, this is probably worth discussing in video call today since I don't have a strong reason to do it, so making this change from the default may be the wrong choice.

Reply:
OK based on direct chat I'm fully convinced that we should match `craft-parts` rather than the `charm` plugin here. Changing...